2.8.0-2026-05-10-dd06e95e-beta compiling errors (ADB Plugin)

```
[ 96%] Building CXX object adb/CMakeFiles/adb.dir/src/ADBShell.cpp.o
[ 96%] Building CXX object multiarc/CMakeFiles/multiarc.dir/src/formats/rar/unrar/sha256.cpp.o
/home/user/far2l/adb/src/ADBPlugin.cpp: In member function ‘std::map<std::__cxx11::basic_string<char>, ADBPlugin::DirMeta> ADBPlugin::PrescanDeviceDirs(const std::vector<std::__cxx11::basic_string<char> >&)’:
/home/user/far2l/adb/src/ADBPlugin.cpp:2423:14: warning: unused variable ‘ms’ [-Wunused-variable]
 2423 |         auto ms = std::chrono::duration_cast<std::chrono::milliseconds>(
      |              ^~
[ 96%] Building CXX object NetRocks/CMakeFiles/NetRocks.dir/src/Protocol/SplitLocationSpecification.cpp.o
[ 96%] Building CXX object multiarc/CMakeFiles/multiarc.dir/src/formats/rar/unrar/filefn.cpp.o
/home/user/far2l/adb/src/ADBPlugin.cpp: In member function ‘void ADBPlugin::GetOpenPluginInfo(OpenPluginInfo*)’:
/home/user/far2l/adb/src/ADBPlugin.cpp:526:41: warning: storing the address of local variable ‘deviceTitles’ in ‘deviceMode.PanelMode::ColumnTitles’ [-Wdangling-pointer=]
  526 |                 deviceMode.ColumnTitles = deviceTitles;
      |                 ~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~
/home/user/far2l/adb/src/ADBPlugin.cpp:507:24: note: ‘deviceTitles’ declared here
  507 |         const wchar_t* deviceTitles[] = { Lng(MColSerial), Lng(MColDeviceName), Lng(MColModel), Lng(MColPort) };
      |                        ^~~~~~~~~~~~
/home/user/far2l/adb/src/ADBPlugin.cpp:494:26: note: ‘deviceMode’ declared here
  494 |         static PanelMode deviceMode = {
      |                          ^~~~~~~~~~
[ 96%] Building CXX object NetRocks/CMakeFiles/NetRocks.dir/src/Protocol/Protocol.cpp.o
[ 96%] Building CXX object adb/CMakeFiles/adb.dir/src/ADBDevice.cpp.o
[ 96%] Linking CXX shared module ../install/Plugins/NetRocks/plug/NetRocks.far-plug-wide
[ 96%] Built target NetRocks
[ 96%] Building CXX object adb/CMakeFiles/adb.dir/src/ADBDialogs.cpp.o
/home/user/far2l/adb/src/ADBDevice.cpp: In member function ‘int ADBDevice::TransferItem(const std::string&, const std::string&, bool, bool, const AdbProgressFn&, const std::function<bool()>&)’:
/home/user/far2l/adb/src/ADBDevice.cpp:626:17: warning: unused variable ‘tail’ [-Wunused-variable]
  626 |     const char* tail = result.c_str() + (result.size() - tail_len);
      |                 ^~~~
[ 96%] Building CXX object multiarc/CMakeFiles/multiarc.dir/src/formats/rar/unrar/filcreat.cpp.o
[ 96%] Building CXX object multiarc/CMakeFiles/multiarc.dir/src/formats/rar/unrar/headers.cpp.o
[ 96%] Building CXX object multiarc/CMakeFiles/multiarc.dir/src/formats/rar/unrar/blake2s.cpp.o
[ 96%] Building CXX object multiarc/CMakeFiles/multiarc.dir/src/formats/rar/unrar/options.cpp.o
[ 96%] Building CXX object adb/CMakeFiles/adb.dir/src/ADBLog.cpp.o
[ 96%] Building CXX object adb/CMakeFiles/adb.dir/src/ProgressBatch.cpp.o
[ 96%] Building CXX object multiarc/CMakeFiles/multiarc.dir/src/formats/rar/unrar/volume.cpp.o
[ 96%] Building CXX object multiarc/CMakeFiles/multiarc.dir/src/formats/rar/unrar/system.cpp.o
[ 96%] Building CXX object multiarc/CMakeFiles/multiarc.dir/src/formats/rar/unrar/crypt.cpp.o
/home/user/far2l/adb/src/ProgressBatch.cpp: In lambda function:
/home/user/far2l/adb/src/ProgressBatch.cpp:215:24: warning: unused variable ‘unit_idx’ [-Wunused-variable]
  215 |                 size_t unit_idx = 0;
      |                        ^~~~~~~~
[ 97%] Linking CXX shared module ../install/Plugins/adb/plug/adb.far-plug-wide
[ 97%] Building CXX object multiarc/CMakeFiles/multiarc.dir/src/formats/rar/unrar/cmddata.cpp.o
Stripping debug symbols from adb plugin
[ 97%] Building CXX object multiarc/CMakeFiles/multiarc.dir/src/formats/rar/unrar/extinfo.cpp.o
[ 97%] Built target adb

```

Я бы не сказал, что что-то сильно изменилось...

2.8.0-2026-05-24-b577d374-beta

```
[ 97%] Building CXX object adb/CMakeFiles/adb.dir/src/ADBShell.cpp.o
/home/far2l/adb/src/ADBPlugin.cpp: In member function ‘std::map<std::__cxx11::basic_string<char>, ADBPlugin::DirMeta> ADBPlugin::PrescanDeviceDirs(const std::vector<std::__cxx11::basic_string<char> >&)’:
/home/far2l/adb/src/ADBPlugin.cpp:2423:14: warning: unused variable ‘ms’ [-Wunused-variable]
 2423 |         auto ms = std::chrono::duration_cast<std::chrono::milliseconds>(
      |              ^~
[ 97%] Building CXX object NetRocks/CMakeFiles/NetRocks.dir/src/Protocol/SplitLocationSpecification.cpp.o
/home/far2l/adb/src/ADBPlugin.cpp: In member function ‘void ADBPlugin::GetOpenPluginInfo(OpenPluginInfo*)’:
/home/far2l/adb/src/ADBPlugin.cpp:526:41: warning: storing the address of local variable ‘deviceTitles’ in ‘deviceMode.PanelMode::ColumnTitles’ [-Wdangling-pointer=]
  526 |                 deviceMode.ColumnTitles = deviceTitles;
      |                 ~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~
/home/far2l/adb/src/ADBPlugin.cpp:507:24: note: ‘deviceTitles’ declared here
  507 |         const wchar_t* deviceTitles[] = { Lng(MColSerial), Lng(MColDeviceName), Lng(MColModel), Lng(MColPort) };
      |                        ^~~~~~~~~~~~
/home/far2l/adb/src/ADBPlugin.cpp:494:26: note: ‘deviceMode’ declared here
  494 |         static PanelMode deviceMode = {
      |                          ^~~~~~~~~~
[ 97%] Building CXX object multiarc/CMakeFiles/multiarc.dir/src/formats/rar/unrar/filefn.cpp.o
[ 97%] Building CXX object NetRocks/CMakeFiles/NetRocks.dir/src/Protocol/Protocol.cpp.o
[ 97%] Building CXX object multiarc/CMakeFiles/multiarc.dir/src/formats/rar/unrar/filcreat.cpp.o
[ 97%] Linking CXX shared module ../install/Plugins/NetRocks/plug/NetRocks.far-plug-wide
[ 97%] Built target NetRocks
[ 97%] Building CXX object adb/CMakeFiles/adb.dir/src/ADBDevice.cpp.o
[ 97%] Building CXX object adb/CMakeFiles/adb.dir/src/ADBDialogs.cpp.o
[ 97%] Building CXX object multiarc/CMakeFiles/multiarc.dir/src/formats/rar/unrar/headers.cpp.o
/home/far2l/adb/src/ADBDevice.cpp: In member function ‘int ADBDevice::TransferItem(const std::string&, const std::string&, bool, bool, const AdbProgressFn&, const std::function<bool()>&)’:
/home/far2l/adb/src/ADBDevice.cpp:626:17: warning: unused variable ‘tail’ [-Wunused-variable]
  626 |     const char* tail = result.c_str() + (result.size() - tail_len);
      |                 ^~~~
[ 97%] Building CXX object multiarc/CMakeFiles/multiarc.dir/src/formats/rar/unrar/blake2s.cpp.o
[ 97%] Building CXX object multiarc/CMakeFiles/multiarc.dir/src/formats/rar/unrar/options.cpp.o
[ 97%] Building CXX object multiarc/CMakeFiles/multiarc.dir/src/formats/rar/unrar/volume.cpp.o
[ 97%] Building CXX object adb/CMakeFiles/adb.dir/src/ADBLog.cpp.o
[ 97%] Building CXX object adb/CMakeFiles/adb.dir/src/ProgressBatch.cpp.o
[ 97%] Building CXX object multiarc/CMakeFiles/multiarc.dir/src/formats/rar/unrar/system.cpp.o
[ 97%] Building CXX object multiarc/CMakeFiles/multiarc.dir/src/formats/rar/unrar/crypt.cpp.o
/home/far2l/adb/src/ProgressBatch.cpp: In lambda function:
/home/far2l/adb/src/ProgressBatch.cpp:215:24: warning: unused variable ‘unit_idx’ [-Wunused-variable]
  215 |                 size_t unit_idx = 0;
      |                        ^~~~~~~~
[ 97%] Building CXX object multiarc/CMakeFiles/multiarc.dir/src/formats/rar/unrar/cmddata.cpp.o
[ 97%] Linking CXX shared module ../install/Plugins/adb/plug/adb.far-plug-wide
Stripping debug symbols from adb plugin
[ 97%] Building CXX object multiarc/CMakeFiles/multiarc.dir/src/formats/rar/unrar/extinfo.cpp.o
[ 97%] Built target adb
```

--- a/adb/src/ADBDevice.cpp
+++ b/adb/src/ADBDevice.cpp
@@ -622,11 +622,13 @@ int ADBDevice::TransferItem(const std::string& src, const std::string& dst, bool
     }
 
     int errno_mapped = Str2Errno(result);
+#if defined(DEBUG) || defined(_DEBUG)
     const size_t tail_len = std::min<size_t>(result.size(), 400);
     const char* tail = result.c_str() + (result.size() - tail_len);
     DBG("TransferItem FAIL is_push=%d pty_exit=%d errno=%d src='%s' dst='%s' tail[%zu]='%.*s'\n",
         is_push, ADBShell::lastPtyExit(), errno_mapped, src.c_str(), dst.c_str(),
         tail_len, (int)tail_len, tail);
+#endif
     return errno_mapped;
 }
 

--- a/adb/src/ADBDevice.cpp
+++ b/adb/src/ADBDevice.cpp
@@ -623,7 +623,7 @@ int ADBDevice::TransferItem(const std::string& src, const std::string& dst, bool
 
     int errno_mapped = Str2Errno(result);
     const size_t tail_len = std::min<size_t>(result.size(), 400);
-    [[maybe_unused]] const char* tail = result.c_str() + (result.size() - tail_len);
+    const char* tail = result.c_str() + (result.size() - tail_len);
     DBG("TransferItem FAIL is_push=%d pty_exit=%d errno=%d src='%s' dst='%s' tail[%zu]='%.*s'\n",
         is_push, ADBShell::lastPtyExit(), errno_mapped, src.c_str(), dst.c_str(),
         tail_len, (int)tail_len, tail);

--- a/adb/src/ADBDevice.cpp
+++ b/adb/src/ADBDevice.cpp
@@ -623,7 +623,7 @@ int ADBDevice::TransferItem(const std::string& src, const std::string& dst, bool
 
     int errno_mapped = Str2Errno(result);
     const size_t tail_len = std::min<size_t>(result.size(), 400);
-    const char* tail = result.c_str() + (result.size() - tail_len);
+    [[maybe_unused]] const char* tail = result.c_str() + (result.size() - tail_len);
     DBG("TransferItem FAIL is_push=%d pty_exit=%d errno=%d src='%s' dst='%s' tail[%zu]='%.*s'\n",
         is_push, ADBShell::lastPtyExit(), errno_mapped, src.c_str(), dst.c_str(),
         tail_len, (int)tail_len, tail);

--- a/adb/src/ADBLog.h
+++ b/adb/src/ADBLog.h
@@ -16,13 +16,13 @@ void DebugLog(const char *format, ...);
 }
 #endif
 
-// Debug macros - completely absent from release builds; enable with -DDEBUG.
+// In release DBG(...) discards its arguments at the preprocessor level, so
+// DBG-only variables can be guarded with `#if defined(DEBUG) || defined(_DEBUG)`
+// at the call site without wrapping the DBG call itself.
 #if defined(DEBUG) || defined(_DEBUG)
 #define DBG(fmt, ...) DebugLog("[%s] " fmt, __FUNCTION__, ##__VA_ARGS__)
 #else
-template <typename... Args>
-constexpr void DebugLogDummy(const char *, Args &&...) noexcept {}
-#define DBG(fmt, ...) (false ? (void)DebugLogDummy((fmt), ##__VA_ARGS__) : (void)0)
+#define DBG(fmt, ...) ((void)0)
 #endif
 
 #endif // ADBLOG_H

--- a/adb/src/ADBLog.h
+++ b/adb/src/ADBLog.h
@@ -16,11 +16,13 @@ void DebugLog(const char *format, ...);
 }
 #endif
 
-// Debug macros - completely absent from release builds; enable with -DDEBUG
+// Debug macros - completely absent from release builds; enable with -DDEBUG.
 #if defined(DEBUG) || defined(_DEBUG)
 #define DBG(fmt, ...) DebugLog("[%s] " fmt, __FUNCTION__, ##__VA_ARGS__)
 #else
-#define DBG(fmt, ...) ((void)0)
+template <typename... Args>
+constexpr void DebugLogDummy(const char *, Args &&...) noexcept {}
+#define DBG(fmt, ...) (false ? (void)DebugLogDummy((fmt), ##__VA_ARGS__) : (void)0)
 #endif
 
 #endif // ADBLOG_H

--- a/adb/src/ADBPlugin.cpp
+++ b/adb/src/ADBPlugin.cpp
@@ -489,7 +489,11 @@ void ADBPlugin::GetOpenPluginInfo(OpenPluginInfo *Info)
 		.Reserved           = {0, 0}
 	};
 
-	const wchar_t* connectedTitles[] = { Lng(MColName), Lng(MColSize) };
+	// Static storage so the pointer stored in `connectedMode.ColumnTitles` outlives this call;
+	// values refreshed on every invocation so language changes are picked up.
+	static const wchar_t* connectedTitles[2];
+	connectedTitles[0] = Lng(MColName);
+	connectedTitles[1] = Lng(MColSize);
 
 	static PanelMode deviceMode = {
 		.ColumnTypes        = L"N,C0,C1,C2",
@@ -504,7 +508,12 @@ void ADBPlugin::GetOpenPluginInfo(OpenPluginInfo *Info)
 		.Reserved           = {0, 0}
 	};
 
-	const wchar_t* deviceTitles[] = { Lng(MColSerial), Lng(MColDeviceName), Lng(MColModel), Lng(MColPort) };
+	// See note above on connectedTitles — same lifetime requirement.
+	static const wchar_t* deviceTitles[4];
+	deviceTitles[0] = Lng(MColSerial);
+	deviceTitles[1] = Lng(MColDeviceName);
+	deviceTitles[2] = Lng(MColModel);
+	deviceTitles[3] = Lng(MColPort);
 
 	if (_isConnected) {
 		connectedMode.ColumnTitles = connectedTitles;
@@ -2410,7 +2419,7 @@ std::map<std::string, ADBPlugin::DirMeta> ADBPlugin::PrescanDeviceDirs(const std
 	auto t0 = std::chrono::steady_clock::now();
 	std::map<std::string, std::unordered_map<std::string, uint64_t>> raw;
 	_adbDevice->BatchDirectoryFileSizes(dirs, raw);
-	uint64_t total_files = 0, total_bytes = 0;
+	[[maybe_unused]] uint64_t total_files = 0, total_bytes = 0;
 	for (auto& kv : raw) {
 		DirMeta dm;
 		dm.file_sizes = std::move(kv.second);
@@ -2420,7 +2429,7 @@ std::map<std::string, ADBPlugin::DirMeta> ADBPlugin::PrescanDeviceDirs(const std
 		total_bytes += dm.total_size;
 		out[kv.first] = std::move(dm);
 	}
-	auto ms = std::chrono::duration_cast<std::chrono::milliseconds>(
+	[[maybe_unused]] auto ms = std::chrono::duration_cast<std::chrono::milliseconds>(
 		std::chrono::steady_clock::now() - t0).count();
 	DBG("PrescanDeviceDirs dirs=%zu out=%zu files=%llu bytes=%llu in %lldms\n",
 		dirs.size(), out.size(), (unsigned long long)total_files,

--- a/adb/src/ADBPlugin.cpp
+++ b/adb/src/ADBPlugin.cpp
@@ -2416,24 +2416,32 @@ std::map<std::string, ADBPlugin::DirMeta> ADBPlugin::PrescanDeviceDirs(const std
 {
 	std::map<std::string, DirMeta> out;
 	if (dirs.empty() || !_isConnected || !_adbDevice) return out;
+#if defined(DEBUG) || defined(_DEBUG)
 	auto t0 = std::chrono::steady_clock::now();
+#endif
 	std::map<std::string, std::unordered_map<std::string, uint64_t>> raw;
 	_adbDevice->BatchDirectoryFileSizes(dirs, raw);
+#if defined(DEBUG) || defined(_DEBUG)
 	uint64_t total_files = 0, total_bytes = 0;
+#endif
 	for (auto& kv : raw) {
 		DirMeta dm;
 		dm.file_sizes = std::move(kv.second);
 		dm.file_count = dm.file_sizes.size();
 		for (const auto& fk : dm.file_sizes) dm.total_size += fk.second;
+#if defined(DEBUG) || defined(_DEBUG)
 		total_files += dm.file_count;
 		total_bytes += dm.total_size;
+#endif
 		out[kv.first] = std::move(dm);
 	}
+#if defined(DEBUG) || defined(_DEBUG)
 	auto ms = std::chrono::duration_cast<std::chrono::milliseconds>(
 		std::chrono::steady_clock::now() - t0).count();
 	DBG("PrescanDeviceDirs dirs=%zu out=%zu files=%llu bytes=%llu in %lldms\n",
 		dirs.size(), out.size(), (unsigned long long)total_files,
 		(unsigned long long)total_bytes, (long long)ms);
+#endif
 	if (out.size() < dirs.size()) {
 		DBG("PrescanDeviceDirs WARN missing %zu dir(s) — totals will fallback to 0/0\n",
 			dirs.size() - out.size());

--- a/adb/src/ADBPlugin.cpp
+++ b/adb/src/ADBPlugin.cpp
@@ -2419,7 +2419,7 @@ std::map<std::string, ADBPlugin::DirMeta> ADBPlugin::PrescanDeviceDirs(const std
 	auto t0 = std::chrono::steady_clock::now();
 	std::map<std::string, std::unordered_map<std::string, uint64_t>> raw;
 	_adbDevice->BatchDirectoryFileSizes(dirs, raw);
-	[[maybe_unused]] uint64_t total_files = 0, total_bytes = 0;
+	uint64_t total_files = 0, total_bytes = 0;
 	for (auto& kv : raw) {
 		DirMeta dm;
 		dm.file_sizes = std::move(kv.second);
@@ -2429,7 +2429,7 @@ std::map<std::string, ADBPlugin::DirMeta> ADBPlugin::PrescanDeviceDirs(const std
 		total_bytes += dm.total_size;
 		out[kv.first] = std::move(dm);
 	}
-	[[maybe_unused]] auto ms = std::chrono::duration_cast<std::chrono::milliseconds>(
+	auto ms = std::chrono::duration_cast<std::chrono::milliseconds>(
 		std::chrono::steady_clock::now() - t0).count();
 	DBG("PrescanDeviceDirs dirs=%zu out=%zu files=%llu bytes=%llu in %lldms\n",
 		dirs.size(), out.size(), (unsigned long long)total_files,

--- a/adb/src/ProgressBatch.cpp
+++ b/adb/src/ProgressBatch.cpp
@@ -212,7 +212,9 @@ BatchResult RunBatch(const std::wstring& title,
 
 	op.Run([&](ProgressState& state) {
 		uint64_t bytes_done = 0, files_done = 0;
+#if defined(DEBUG) || defined(_DEBUG)
 		size_t unit_idx = 0;
+#endif
 		try {
 			for (auto& u : units) {
 				if (state.ShouldAbort()) { result.aborted = true; break; }

--- a/adb/src/ProgressBatch.cpp
+++ b/adb/src/ProgressBatch.cpp
@@ -212,7 +212,7 @@ BatchResult RunBatch(const std::wstring& title,
 
 	op.Run([&](ProgressState& state) {
 		uint64_t bytes_done = 0, files_done = 0;
-		[[maybe_unused]] size_t unit_idx = 0;
+		size_t unit_idx = 0;
 		try {
 			for (auto& u : units) {
 				if (state.ShouldAbort()) { result.aborted = true; break; }

--- a/adb/src/ProgressBatch.cpp
+++ b/adb/src/ProgressBatch.cpp
@@ -212,7 +212,7 @@ BatchResult RunBatch(const std::wstring& title,
 
 	op.Run([&](ProgressState& state) {
 		uint64_t bytes_done = 0, files_done = 0;
-		size_t unit_idx = 0;
+		[[maybe_unused]] size_t unit_idx = 0;
 		try {
 			for (auto& u : units) {
 				if (state.ShouldAbort()) { result.aborted = true; break; }

